src/parser/shared/modules/Combatants.ts

```diff
@@ -54,7 +54,11 @@ class Combatants extends Entities<Combatant> {
         this.players[combatantInfo.sourceID] = new Combatant(this.owner, combatantInfo);
       }
     });
-    this._selected = this.players[this.owner.playerId];
+    const selected = this.players[this.owner.playerId];
+    if (selected != null && !validSelectedCombatant(selected)) {
```

Comment:
```suggestion
    if (selected && !validSelectedCombatant(selected)) {
```

can be simplified to this to avoid the double-equals check (immediately drew my eye because those are *usually* a bad thing but it does actually do useful things here by passing on null *and* undefined)

Reply:
I prefer to explicitly check for `undefined` and `null` rather than just check for truthy. `a == null` is true for `undefined` and `null` and it's clear to the reader that we're checking that it's defined rather than checking for a truthy value. But saying that, this is no hill I plan to expire upon